### DISCUSSION TITLE
There are differences between `.ports` and `pprint_ports()`

### DISCUSSION THREAD

**Discussion by maxxxxwell** [gdsfactory/gdsfactory#3326] [Q&A] [answered]
Hi
I found this problem , Any one can help me 
Thanks 
1:
```python
c = gf.Component()

cross_section = gf.cross_section.cross_section(width=0.5,layer=[2,1])
s1 = c << gf.components.straight(length=10,cross_section=cross_section)

print(s1.ports)

c.add_ports(s1.ports)
c.pprint_ports()
```
![port3](https://github.com/user-attachments/assets/d1ecadda-a688-4df7-ac5e-87d6f112b8b9)

2:
```python
cross_section = gf.cross_section.cross_section(width=0.5,layer=[2,1])
c = gf.components.straight(length=10,cross_section=cross_section)
print(c.ports)
c.pprinr_ports()
```
![port4](https://github.com/user-attachments/assets/802acd05-2c3b-4252-b2bf-62f041efdeaa)

**Comment by sebastian-goeldi** [comment, marked as answer]
Yes, one is the layer index and one the actual layer definition.


In klayout internal terms, `pprint_ports` gives you the LayerInfo, i.e. the gds representation of a layer/datatype pair. `port.layer`, which is what `.ports` prints is the layer index, is an internal number used by klayout to uniquely identify a layer within a layout object.

We can change to string representation of ports to reflect that and report the layer info instead of the internal index.

**Reply by maxxxxwell** [reply]
Thanks for your information
